tensorflow-frameworks/tensorflow-data/src/main/java/org/tensorflow/data/Utils.java

```diff
@@ -0,0 +1,34 @@
+package org.tensorflow.data;
+
+import org.tensorflow.tools.Shape;
+
+public class Utils {
+
+    private static Shape head(Shape shape) {
+        return Shape.of(shape.size(0));
+    }
+
+    public static Shape tail(Shape shape) {
+        long[] tail = new long[shape.numDimensions() - 1];
+        for (int i = 1; i < shape.numDimensions(); i++) {
+            tail[i - 1] = shape.size(i);
+        }

```

Comment:
That could be replaced by `Arrays.copyOfRange(shape.asArray(), 1, shape.numDimensions)`

Reply:
Yup, I'm using `copyOfRange` in the `Shape.tail` implementation.